lib/std/fs/test.zig

```diff
@@ -745,6 +745,7 @@ test "directory operations on files" {
 test "file operations on directories" {
     // TODO: fix this test on FreeBSD. https://github.com/ziglang/zig/issues/1759
     if (native_os == .freebsd) return error.SkipZigTest;
+    if (native_os == .wasi) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/20747
```

Comment:
```suggestion
    if (native_os == .wasi and builtin.link_libc) return error.SkipZigTest; // https://github.com/ziglang/zig/issues/20747
```

This is a bug in the interaction between wasi-libc and wasmtime. The test case works fine with the latest wasmtime versions when not linking libc